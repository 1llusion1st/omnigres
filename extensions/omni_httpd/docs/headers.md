# Headers

HTTP requests come with headers, which can be retrieved using `omni_http.header_get` and `omni_http.header_get_all`
functions:

```sql
select omni_http.http_header_get(request.headers, 'host') as host;
select omni_http.http_header_get_all(request.headers, 'accept') as accept;
```

The header name these functions take is case-insensitive.