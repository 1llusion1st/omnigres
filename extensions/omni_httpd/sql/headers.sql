select omni_httpd.http_header_get(array [omni_httpd.http_header('Host', 'omnihost')], 'Host');
select omni_httpd.http_header_get(array [omni_httpd.http_header('Host', 'omnihost')], 'host');
select
    omni_httpd.http_header_get_all(
            array [omni_httpd.http_header('Accept', 'application/xml'), omni_httpd.http_header('Accept', 'application/json')],
            'accept');
