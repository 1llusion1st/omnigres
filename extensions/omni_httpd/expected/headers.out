select omni_httpd.http_header_get(array [omni_httpd.http_header('Host', 'omnihost')], 'Host');
 http_header_get 
-----------------
 omnihost
(1 row)

select omni_httpd.http_header_get(array [omni_httpd.http_header('Host', 'omnihost')], 'host');
 http_header_get 
-----------------
 omnihost
(1 row)

select
    omni_httpd.http_header_get_all(
            array [omni_httpd.http_header('Accept', 'application/xml'), omni_httpd.http_header('Accept', 'application/json')],
            'accept');
 http_header_get_all 
---------------------
 application/xml
 application/json
(2 rows)

